src/llama-kv-cache-hybrid-recurrent.cpp

```diff
@@ -32,7 +32,7 @@ llama_kv_cache_hybrid_recurrent::llama_kv_cache_hybrid_recurrent(
     kv_attn(new llama_kv_cache_unified(
         model,
         attn_filter == nullptr ?
-            [&](int32_t il) { return !model.hparams.recurrent_layer(il); }
+            [&](int32_t il) { return model.hparams.recurrent_layer(il); }
```

Comment:
The only open question I have on my end is how we can refactor this so that it can also accept models where we have attn and ssm - do you have any idea how we can tackle this @gabe-l-hart ? 👀